plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -298,15 +298,19 @@ private static ConnectorPageSource createOrcPageSource(
                         .collect(Collectors.groupingBy(
                                 HiveColumnHandle::getBaseColumnName,
                                 mapping(
-                                        column -> column.getHiveColumnProjectionInfo().map(HiveColumnProjectionInfo::getDereferenceNames).orElse(ImmutableList.<String>of()),
+                                        column -> column.getHiveColumnProjectionInfo()
+                                                .map(info -> info.getDereferenceNames().stream().map(dereference -> dereference.toLowerCase(ENGLISH)).collect(toList()))
+                                                .orElse(ImmutableList.<String>of()),
                                         toList())));
             }
             else {
                 projectionsByColumnIndex = projections.stream()
                         .collect(Collectors.groupingBy(
                                 HiveColumnHandle::getBaseHiveColumnIndex,
                                 mapping(
-                                        column -> column.getHiveColumnProjectionInfo().map(HiveColumnProjectionInfo::getDereferenceNames).orElse(ImmutableList.<String>of()),
+                                        column -> column.getHiveColumnProjectionInfo()
```

Comment:
since this is now getting a bit longer at two places, we could extract it out to simplify. (I'd leave it up to you if you'd like to tackle it separately.)

```
    projectionsByColumnName = projections.stream()
            .collect(Collectors.groupingBy(
                        HiveColumnHandle::getBaseColumnName,
                        mapping(OrcPageSourceFactory::getDereferencesAsList, toList())));


    private static List<String> getDereferencesAsList(HiveColumnHandle column)
    {
        return column.getHiveColumnProjectionInfo()
                .map(info -> info.getDereferenceNames().stream()
                        .map(dereference -> dereference.toLowerCase(ENGLISH))
                        .collect(toList()))
                .orElse(ImmutableList.<String>of());
    }
```